src/Data/Variant.purs

```diff
@@ -73,6 +79,70 @@ on p f g r =
   coerceR ∷ Variant r2 → Variant r1
   coerceR = unsafeCoerce
 
+-- | Type class that matches a row for a `record` that will eliminate a row for
+-- | a `variant`, producing a `result` of the specified type. Just a wrapper for
+-- | `VariantMatch` to convert `RowToList` and vice versa.
+class VariantRecordElim
+    (variant ∷ # Type)
+    (record ∷ # Type)
+    result
+  | variant result → record
+  , record → variant result
```

Comment:
Actually, this type class could have `| -> variant record result`, but the other one should have `| vlist -> rlist result, rlist -> vlist result`.